datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreFlowConfig.java

```diff
@@ -0,0 +1,133 @@
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_DATABASE_RESTORE_IN_PROGRESS_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_DATABASE_RESTORE_START_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_FULL_RESTORE_IN_PROGRESS_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_RESTORE_FAILED_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_RESTORE_FINISHED_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.DATALAKE_TRIGGERING_RESTORE_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.FINAL_STATE;
+import static com.sequenceiq.datalake.flow.dr.restore.DatalakeRestoreState.INIT_STATE;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.flow.core.config.AbstractFlowConfiguration;
+import com.sequenceiq.flow.core.config.RetryableFlowConfiguration;
+
+@Component
+public class DatalakeRestoreFlowConfig extends AbstractFlowConfiguration<DatalakeRestoreState, DatalakeRestoreEvent>
+        implements RetryableFlowConfiguration<DatalakeRestoreEvent> {
+
+    private static final List<Transition<DatalakeRestoreState, DatalakeRestoreEvent>> TRANSITIONS =
+            new Transition.Builder<DatalakeRestoreState, DatalakeRestoreEvent>()
+                    .defaultFailureEvent(DATALAKE_DATABASE_RESTORE_FAILED_EVENT)
+
+                    .from(INIT_STATE)
+                    .to(DATALAKE_TRIGGERING_RESTORE_STATE)
+                    .event(DATALAKE_TRIGGER_RESTORE_EVENT).noFailureEvent()
+
+                    .from(DATALAKE_TRIGGERING_RESTORE_STATE)
+                    .to(DATALAKE_DATABASE_RESTORE_START_STATE)
+                    .event(DATALAKE_DATABASE_RESTORE_EVENT)
+                    .failureState(DATALAKE_RESTORE_FAILED_STATE)
+                    .failureEvent(DATALAKE_RESTORE_FAILED_EVENT)
+
+                    .from(INIT_STATE)
+                    .to(DATALAKE_DATABASE_RESTORE_START_STATE)
+                    .event(DATALAKE_DATABASE_RESTORE_EVENT).noFailureEvent()
+
+                    .from(DATALAKE_DATABASE_RESTORE_START_STATE)
+                    .to(DATALAKE_DATABASE_RESTORE_IN_PROGRESS_STATE)
+                    .event(DATALAKE_DATABASE_RESTORE_IN_PROGRESS_EVENT)
+                    .failureState(DATALAKE_DATABASE_RESTORE_COULD_NOT_START_STATE)
+                    .failureEvent(DATALAKE_DATABASE_RESTORE_COULD_NOT_START_EVENT)
+
+                    .from(DATALAKE_DATABASE_RESTORE_COULD_NOT_START_STATE)
+                    .to(DATALAKE_FULL_RESTORE_IN_PROGRESS_STATE)
+                    .event(DATALAKE_DATABASE_RESTORE_FAILURE_HANDLED_EVENT)
+                    .failureState(DATALAKE_FULL_RESTORE_IN_PROGRESS_STATE)
+                    .failureEvent(DATALAKE_DATABASE_RESTORE_FAILED_EVENT)
```

Comment:
could you describe what this part does please?

Reply:
if database cannot start or failed, it will go to datalake backup in progress status which pulls the thunderhead backup status. 

datalake backup include 1. database which is done by cloudbreak. 2. solr, hms which is done by thunderhead. So even database backup cannot start or failed, it should go to the datalake thunderhead pulling and finish part 2.  part 1 and part 2 are independent. 

Please notice this part is not my code. 

this DatalakeRestoreFlowConfig.java is renamed from DatalakeDatabaseRestoreFlowConfig.java

So the specific code you see is the previous there working code in prod from Kalyan. If you need more explanation. I will get Kalyan to answer more here.

Reply:
@kkalvagadda1